examples/repl.rs

```diff
@@ -58,6 +61,67 @@ mod example {
         disassembler: DisassemblySource,
     }
 
+    /// The subcommands that are acceptable for backtrace.
+    enum BacktraceType {
+        // uses the frame_pointer_unwinder.
+        FramePtr,
+
+        // uses naive_unwinder.
+        Naive,
+    }
+
+    impl BacktraceType {
+        fn from_str(value: &str) -> CrabResult<Self> {
+            match value {
+                "fp" | "" => Ok(BacktraceType::FramePtr),
+                "naive" => Ok(BacktraceType::Naive),
+                _ => {
+                        Err(format!("Unrecognized backtrace type {}. Supported ones are 'fp' and 'naive'. Please consider using one of them.", value).into())
+                    },
+            }
+        }
+    }
+
+    impl Default for BacktraceType {
+        fn default() -> Self {
+            BacktraceType::FramePtr
+        }
+    }
+
+    impl HighlightAndComplete for BacktraceType {
+        type Error = VoidError;
+        fn from_str(line: &str) -> Result<Self, Self::Error> {
+            Ok(BacktraceType::from_str(line.trim()).unwrap_or_default())
```

Comment:
Could you inline this call and actually return the error when there is any? You could use `type Error = String` above.